charts/spire/charts/spire-server/templates/hpa.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2
```

Comment:
A thought...

Since this file came about, we added a check here:
https://github.com/spiffe/helm-charts-hardened/blob/main/charts/spire/charts/spire-server/templates/statefulset.yaml#L19-L21

to ensure scaleup doesn't happen when the database type doesn't support scaling. Could you add a similar check (maybe just fail outright if sqlite & autoscaling enabled) between line 1 and 2 please?

Reply:
There is an existing check in hpa.yaml: https://github.com/spiffe/helm-charts-hardened/blob/spire-0.16.0/charts/spire/charts/spire-server/templates/hpa.yaml#L15
It should fail if sqlite is selected and the number of replicas is greater than 1.